lib/db/seeder.rb

```diff
@@ -0,0 +1,200 @@
+# frozen_string_literal: true
+# Copyright (c) 2015 The Regents of the University of Michigan.
+# All Rights Reserved.
+# Licensed according to the terms of the Revised BSD License
+# See LICENSE.md for details.
+
+require "json"
+
+module Db
+
+  # Create domain objects from seed hashes if they do not already exist.
+  # This class does _not_ update objects that already exist.
+  #
+  # A log entry is created for each seed entry that is created,
+  # each skipped entry (if it already exists), and each entry that
+  # the seeder failed to create.
+  class Seeder
+
+    # @param logger [Logger] logger  Defaults to the Rails.logger
+    def initialize(logger = nil)
+      @logger = logger || Rails.logger
+      @maps = {
+        fixity_algs:      [FixityAlg,     proc {|e| e[:name] },       proc {|e| e.slice(:name) }],
+        protocols:        [Protocol,      proc {|e| e[:name] },       proc {|e| e.slice(:name) }],
+        storage_regions:  [StorageRegion, proc {|e| e[:name] },       proc {|e| e.slice(:name) }],
+        storage_types:    [StorageType,   proc {|e| e[:name] },       proc {|e| e.slice(:name) }],
+        members:          [Member,        proc {|e| e[:name] },       proc {|e| member_from_entry(e) }],
+        nodes:            [Node,          proc {|e| e[:namespace] },  proc {|e| node_from_entry(e) }]
+      }
+    end
+
+    # Create the objects described by the entries hash, of the given
+    # type, if they do not already exist.
+    # @overload create(type, entries)
+    #   @param type [Symbol] Plural type, corresponding to the class of the
+    #     record to be created as well as the field in the seeds file.  Includes
+    #     :fixity_algs, :protocols, :storage_regions, :storage_types.
+    #   @param entries [Array<String>] The names of the records to be created.
+    #
+    # @overload create(type, entries)
+    #   @param type [Symbol] Must be :members
+    #   @param entries [Array<Hash>] Array of entries
+    #   @option entries [String] :member_id
+    #   @option entries [String] :name
+    #   @option entries [String] :email
+    #
+    # @overload create(type, entries)
+    #   @param type [Symbol] Must be :nodes
+    #   @param entries [Array<Hash>] Array of entries.
+    #   @option entries [String] :namespace
+    #   @option entries [String] :name
+    #   @option entries [String] :storage_region StorageRegion name
+    #   @option entries [String] :storage_type StorageType name
+    #   @option entries [Array<String>] :fixity_algs FixityAlg names
+    #   @option entries [Array<String>] :protocols Protocol names
+    #   @option entries [String] :api_root
+    #   @option entries [String] :private_auth_token Unencrypted
+    #   @option entries [String] :auth_credential Unencrypted
+    def create(type, entries)
+      create_from_entries(entries, *maps[type])
+    end
```

Comment:
* Re: putting the details into code directly: 
  Seeder already expects a ruby hash, not JSON.  Representing the data in a `.rb` file instead of .`json` would be fine, but doesn't gain us anything.

* Re: json-schema to validate the seeds
That would be more robust, but is overkill for a set of files that are changed so rarely.

* Re: ActiveRecord::Serialization
  The ruby builtin parser was sufficient for my needs.  The AR serializer expects json strings, meaning the seeder is dependent on the serialization format of the data files anyway.